pandas/core/common.py

```diff
@@ -381,25 +382,17 @@ def _asarray_tuplesafe(values, dtype=None):
         return values.values
 
     if isinstance(values, list) and dtype in [np.object_, object]:
-        return lib.list_to_object_array(values)
+        return construct_1d_array_from_listlike(values)
 
     result = np.asarray(values, dtype=dtype)
 
     if issubclass(result.dtype.type, compat.string_types):
         result = np.asarray(values, dtype=object)
 
     if result.ndim == 2:
-        if isinstance(values, list):
-            return lib.list_to_object_array(values)
-        else:
-            # Making a 1D array that safely contains tuples is a bit tricky
-            # in numpy, leading to the following
-            try:
-                result = np.empty(len(values), dtype=object)
-                result[:] = values
-            except ValueError:
-                # we have a list-of-list
-                result[:] = [tuple(x) for x in values]
+        # Avoid building an array of arrays:
```

Comment:
is there an asv that hits this case?

Reply:
I don't even think there is any valid code path that hits this case... which indeed it should be suppressed in #18626

Reply:
if there is not valid code, then let's remove it. or make a new issue. *inside* a PR doesn't help.

Reply:
#18819

Reply:
can you add the issue number here with a TODO

Reply:
(done)